mlflow/server/auth/__init__.py

```diff
@@ -71,45 +396,164 @@ def _before_request():
+    if validator := BEFORE_REQUEST_VALIDATORS.get((request.path, request.method)):
+        _logger.debug(f"Calling validator: {validator.__name__}")
+        if not validator():
+            return make_forbidden_response()
+    else:
+        _logger.debug(f"No validator found for {(request.path, request.method)}")
+
+
+def set_can_manage_experiment_permission(resp: Response):
+    response_message = CreateExperiment.Response()
+    parse_dict(resp.json, response_message)
+    experiment_id = response_message.experiment_id
+    username = request.authorization.username
+    store.create_experiment_permission(experiment_id, username, MANAGE.name)
+
+
+def set_can_manage_registered_model_permission(resp: Response):
+    response_message = CreateRegisteredModel.Response()
+    parse_dict(resp.json, response_message)
+    name = response_message.registered_model.name
+    username = request.authorization.username
+    store.create_registered_model_permission(name, username, MANAGE.name)
+
+
+AFTER_REQUEST_PATH_HANDLERS = {
+    CreateExperiment: set_can_manage_experiment_permission,
+    CreateRegisteredModel: set_can_manage_registered_model_permission,
+}
+
+
+def get_after_request_handler(request_class):
+    return AFTER_REQUEST_PATH_HANDLERS.get(request_class)
+
+
+AFTER_REQUEST_HANDLERS = {
+    (http_path, method): handler
+    for http_path, handler, methods in get_endpoints(get_after_request_handler)
+    for method in methods
+}
+
 
+def _after_request(resp: Response):
+    _logger.debug(f"after_request: {request.method} {request.path}")
+    if 400 <= resp.status_code < 600:
+        return resp
 
-def _after_request(resp):
-    # TODO: Implement post-request logic
+    if handler := AFTER_REQUEST_HANDLERS.get((request.path, request.method)):
+        _logger.debug(f"Calling after request handler: {handler.__name__}")
+        handler(resp)
     return resp
 
 
+def create_admin_user(username, password):
+    if not store.has_user(username):
+        store.create_user(username, password, is_admin=True)
+        _logger.info(
+            f"Created admin user '{username}'. "
+            "It is recommended that you set a new password as soon as possible "
+            f"on {ROUTES.UPDATE_USER_PASSWORD}."
+        )
+
+
+def alert(href: str):
+    return render_template_string(
+        r"""
+<script type = "text/javascript">
+{% with messages = get_flashed_messages() %}
+  {% if messages %}
+    {% for message in messages %}
+      alert("{{ message }}");
+    {% endfor %}
+  {% endif %}
+{% endwith %}
+      window.location.href = "{{ href }}";
+</script>
+""",
+        href=href,
+    )
+
+
 def signup():
-    # TODO: add css
     return render_template_string(
         r"""
+<style>
+  form {
+    background-color: #F5F5F5;
+    border: 1px solid #CCCCCC;
+    border-radius: 4px;
+    padding: 20px;
+    max-width: 400px;
+    margin: 0 auto;
+    font-family: Arial, sans-serif;
+    font-size: 14px;
+    line-height: 1.5;
+  }
```

Comment:
ChatGPT wrote this CSS

Reply:
<img width="767" alt="image" src="https://user-images.githubusercontent.com/17039389/234789317-074b1eb2-cbb0-4d34-ba9b-fbae2d6aaf21.png">

Reply:
looking great!